### PR TITLE
[nplb] Add tests for vertical video

### PR DESCRIPTION
**Pull request by osagie98**
b/248859788

### CODE REVIEW

File: starboard/nplb/player_test_util.cc [hunk 1]
```diff
@@ -147,6 +147,44 @@ std::vector<SbPlayerTestConfig> GetSupportedSbPlayerTestConfigs(
   return test_configs;
 }
 
+std::vector<SbPlayerTestConfig> GetVerticalVideoTestConfigs() {
+  const char* kEmptyName = NULL;
+
+  // TODO: Add these files.
+  const char* kFilenames[] = {"1080_1920_avc.dmp", "1440_2560_av1.dmp",
+                              "3840_2160_av1.dmp"};
```
Comment (jasonzhangxx) [inline starboard/nplb/player_test_util.cc:155]: Is it intended to add "3840_2160_av1.dmp" here?

And one question about why we choose 1080p avc, 2k and 4k av1, with no vp9?

File: starboard/nplb/player_test_util.cc [hunk 2]
```diff
@@ -147,6 +147,44 @@ std::vector<SbPlayerTestConfig> GetSupportedSbPlayerTestConfigs(
...
+
+  const SbPlayerOutputMode kOutputModes[] = {kSbPlayerOutputModeDecodeToTexture,
+                                             kSbPlayerOutputModePunchOut};
+
+  std::vector<const char*> video_files;
+  video_files.push_back(kEmptyName);
```
Comment (jasonzhangxx) [inline starboard/nplb/player_test_util.cc:161]: I think we don't need to test empty video file here.

File: starboard/nplb/vertical_video_test.cc [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+class VerticalVideoTest : public ::testing::TestWithParam<SbPlayerTestConfig> {
+};
+
+TEST(VerticalVideoTest, CapabilityQuery) {
+  // TODO: Check for capability support based on HW reqs.
+}
+
+TEST_P(VerticalVideoTest, WriteSamples) {
+  // TODO: Write samples and check for no playback errors. Potentially try
+  // seeking in a third test?
```
Comment (jasonzhangxx) [inline starboard/nplb/vertical_video_test.cc:38]: I suggest to let player play for 200ms.

File: starboard/nplb/player_test_util.cc (hunk 1 above)
Comment (osagie98) [reply starboard/nplb/player_test_util.cc]: This will change to: 1080p avc, 2k vp9, 4k vp9, and 8k av1. The 4K one should be "2160_3840.dmp"

File: starboard/nplb/player_test_util.cc (hunk 2 above)
Comment (osagie98) [reply starboard/nplb/player_test_util.cc]: Done

File: starboard/nplb/vertical_video_test.cc (hunk 3 above)
Comment (osagie98) [reply starboard/nplb/vertical_video_test.cc]: Done, it writes 6 frames to the player

File: starboard/nplb/vertical_video_test.cc [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                       << " should be supported on this platform.";
+        }
+      };
+
+  SbMediaSupportType support = SbMediaCanPlayMimeAndKeySystem(
+      "video/mp4; codecs=\"avc1.4d002a\"; width=1920; height=1080", "");
+  if (support == kSbMediaSupportTypeProbably) {
+    support = SbMediaCanPlayMimeAndKeySystem(
+        "video/mp4; codecs=\"avc1.4d002a\"; width=608; height=1080", "");
+    check_vertical_resolution_support(support, "608", "1080");
```
Comment (osagie98) [inline starboard/nplb/vertical_video_test.cc:47]: As the vertical resolution requirement is not yet enforced, we don't need to fail if a given resolution is unsupported. I may update the log as technically the resolution doesn't have to be supported.

**Review by jasonzhangxx — COMMENTED** [review]
Please run the tests and filter failed tests on some platforms.

File: starboard/nplb/player_test_util.cc (hunk 1 above)
Comment (osagie98) [reply starboard/nplb/player_test_util.cc]: For now I've added 1080p avc, 4k vp9, and 8k av1. I'll generate and add 2k vp9 sometime later.

**Comment by osagie98** [comment]
nplb test fails due to an unrelated flaked test on win32

File: starboard/nplb/vertical_video_test.cc [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  return test_configs;
+}
+
+TEST(VerticalVideoTest, CapabilityQuery) {
+  SbMediaSupportType support = SbMediaCanPlayMimeAndKeySystem(
+      "video/mp4; codecs=\"avc1.4d002a\"; width=1920; height=1080", "");
+  if (support == kSbMediaSupportTypeProbably) {
+    support = SbMediaCanPlayMimeAndKeySystem(
+        "video/mp4; codecs=\"avc1.4d002a\"; width=608; height=1080", "");
+    CheckVerticalResolutionSupport(support, "608", "1080");
```
Comment (jasonzhangxx) [inline starboard/nplb/vertical_video_test.cc:96]: After considering it again, I think we can change the function to
```
void CheckVerticalVideoSupport(const char* mime) {
  SB_LOG_IF(WARNING,  SbMediaCanPlayMimeAndKeySystem(mime, "") != kSbMediaSupportTypeProbably)
      << "Vertical video (" << mime << ") should be supported on this platform.";
}
```

**Review by jasonzhangxx — APPROVED** [review]
